[TensorExpr] Postpone insertion of Alloc/Free statements in computeAt.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#36526 [TensorExpr] Postpone insertion of Alloc/Free statements in computeAt.**

Differential Revision: [D21004740](https://our.internmc.facebook.com/intern/diff/D21004740)

Commit:
[TensorExpr] Postpone insertion of Alloc/Free statements in computeAt.

[ghstack-poisoned]

bwasti commented on torch/csrc/jit/tensorexpr/loopnest.cpp:
```diff
@@ -1240,6 +1240,11 @@ void LoopNest::computeAt(Stmt* s, For* f) {
+
+  // Mark the new temp buffer as requiring an alloc (it will be inserted as a
+  // part of prepareForCodegen.
+  temp_bufs_.push_back(temp_buf);
```
nit: can we have this is as a single data structure? `std::vector<std::pair<..>>`

bwasti commented on torch/csrc/jit/tensorexpr/loopnest.cpp:
```diff
@@ -1240,6 +1240,11 @@ void LoopNest::computeAt(Stmt* s, For* f) {
   }
+
+  // Mark the new temp buffer as requiring an alloc (it will be inserted as a
+  // part of prepareForCodegen.
```
it might make sense to move the bounds inference to prepareforcodegen

ZolotukhinM replied:
Yes, that's what we should be ultimately doing.

Commit:
Update on "[TensorExpr] Postpone insertion of Alloc/Free statements in computeAt."


Differential Revision: [D21004740](https://our.internmc.facebook.com/intern/diff/D21004740)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Postpone insertion of Alloc/Free statements in computeAt."


Differential Revision: [D21004740](https://our.internmc.facebook.com/intern/diff/D21004740)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Postpone insertion of Alloc/Free statements in computeAt."


Differential Revision: [D21004740](https://our.internmc.facebook.com/intern/diff/D21004740)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Postpone insertion of Alloc/Free statements in computeAt."


Differential Revision: [D21004740](https://our.internmc.facebook.com/intern/diff/D21004740)

[ghstack-poisoned]
